ghga_validator/plugins/ref_validation.py

```diff
@@ -0,0 +1,185 @@
+        """Return the range class for a slot
+
+        Args:
+            slot_def (SlotDefinition): Slot Definition
+
+        Returns:
+            Optional[str]: Range class for a slot
+        """
+        all_classes_name = self.schemaview.all_classes().keys()
+        if slot_def:
+            range_class = slot_def.range
+            if range_class in all_classes_name:
+                return range_class
+        return None
+
+    def get_all_class_ids(self, obj: Dict, target_class: str) -> Dict[str, List[str]]:
+        """Get all lists of identifies of inlined objects organized by class name
+
+        Args:
+            obj (Dict): The object to be parsed
+            target_class (str): Target class
+
+        Returns:
+            Dict[class_name, List[str]]: The dictionary containing the lists of
+            identifiers by the class name
+        """
+        all_ids = {}
+
+        for class_name, identifier, _, _ in ObjectIterator(
+            self.schemaview, obj, target_class
+        ):
+            if class_name not in all_ids:
+                all_ids[class_name] = [identifier]
+            else:
+                all_ids[class_name].append(identifier)
+
+        return all_ids
+
+    def validate_refs(
+        self,
+        object_to_validate: Dict,
+        target_class: str,
+        all_class_ids: Dict,
+    ) -> List[ValidationMessage]:
+        """
+        Validate non inlined reference fields in the JSON data
+
+        Args:
+            object_to_validate: input data
+            target_class: parent class in the schema
+            all_class_ids: pre-computed dictionary containing all identifiers ordered by class
+
+        Returns:
+            List[ValidationMessage]: List of validation messages
+
+        """
+        messages = []
+
+        for class_name, _, data, path in ObjectIterator(
+            self.schemaview, object_to_validate, target_class
+        ):
+            for field, value in data.items():
+                slot_def = self.schemaview.induced_slot(field, class_name)
+                range_class = self.get_range_class(slot_def)
+                if range_class and not self.schemaview.is_inlined(slot_def):
+                    non_match = self.find_missing_refs(
+                        value, all_class_ids[range_class]
+                    )
+                    if len(non_match) == 0:
+                        continue
+                    message = ValidationMessage(
+                        severity=SeverityEnum.error,
+                        message="Unknown references "
+                        + f"({self.non_match_as_string(non_match)})",
+                        field=f"{path_as_string(path)}.{field}",
+                        value=value,
+                    )
+                    messages.append(message)
+        return messages
+
+    def find_missing_refs(
+        self, ref_value: Union[List[str], str], id_list: List
```

Comment:
```suggestion
        self, ref_value: Union[List[Union[Number, str]], Union[Number,str]], id_list: List
```
All types in LinkML can be identifiers

Reply:
Agree, string is used as identifier type for the submission, but it should of course have different type in the generic implementation.